File: app/src/test/java/com/starsep/sokoban/release/activity/ChooseLevelActivityTest.java
package com.starsep.sokoban.release.activity;

import android.content.Intent;
import android.view.View;
import android.widget.Button;
import android.widget.GridView;

import com.starsep.sokoban.release.BuildConfig;
import com.starsep.sokoban.release.R;
import com.starsep.sokoban.release.activity.GameActivity;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.Shadows;
import org.robolectric.annotation.Config;

import static android.view.View.VISIBLE;
import static junit.framework.TestCase.assertTrue;
import static org.junit.Assert.assertEquals;


@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class)
public class ChooseLevelActivityTest {
    private ChooseLevelActivity activity;

    @Before
    public void setUp() throws Exception {
        activity = Robolectric.buildActivity(ChooseLevelActivity.class)
                .create()
                .resume()
                .get();
    }

    @Test
    public void gridViewShouldBeVisible() {
        GridView gridView = activity.findViewById(R.id.gridView);
        assertEquals(VISIBLE, gridView.getVisibility());
    }

    @Test
    public void gridViewShouldHaveLevelButtons() {
        GridView gridView = activity.findViewById(R.id.gridView);
        final int numberOfLevels = activity.getResources().getInteger(R.integer.number_of_levels);
        assertEquals(numberOfLevels, gridView.getCount());
        for (int i = 0; i < numberOfLevels; i++) {
            View view = gridView.getAdapter().getView(i, null, gridView);
            assertTrue(view instanceof Button);
        }
    }

    @Test
    public void levelButtonLaunchesGameActivity() {
        GridView gridView = activity.findViewById(R.id.gridView);
        Button levelButton = (Button) gridView.getAdapter().getView(0, null, gridView);
        levelButton.performClick();
        Intent intent = Shadows.shadowOf(activity).peekNextStartedActivity();
        assertEquals(GameActivity.class.getCanonicalName(), intent.getComponent().getClassName());
    }
}

File: app/src/test/java/com/starsep/sokoban/release/activity/ChooseLevelActivityTest.kt
package com.starsep.sokoban.release.activity

// @RunWith(RobolectricTestRunner::class)
class ChooseLevelActivityTest : RobolectricTest() {
    // TODO: fix
    /*private var activity: MainActivity? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        activity = Robolectric.buildActivity(MainActivity::class.java)
            .create()
            .resume()
            .get()
    }

    @After
    fun after() {
        close()
    }

    @Test
    fun gridViewShouldBeVisible() {
        val gridView = activity!!.findViewById<GridView>(R.id.gridView)
        Assert.assertEquals(
            View.VISIBLE.toLong(),
            gridView.visibility.toLong()
        )
    }

    @Test
    fun gridViewShouldHaveLevelButtons() {
        val gridView = activity!!.findViewById<GridView>(R.id.gridView)
        val numberOfLevels = activity!!.resources.getInteger(R.integer.number_of_levels)
        Assert.assertEquals(numberOfLevels.toLong(), gridView.count.toLong())
        for (i in 0 until numberOfLevels) {
            val view = gridView.adapter.getView(i, null, gridView)
            TestCase.assertTrue(view is Button)
        }
    }

    @Test
    fun levelButtonLaunchesGameActivity() {
        val gridView = activity!!.findViewById<GridView>(R.id.gridView)
        val levelButton =
            gridView.adapter.getView(0, null, gridView) as Button
        levelButton.performClick()
        val intent = Shadows.shadowOf(activity).peekNextStartedActivity()
        Assert.assertEquals(
            GameFragment::class.java.canonicalName,
            intent.component!!.className
        )
    }*/
}
